.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,51 @@
+name: "Continuous Integration"
+on:
+  push:
+    # run on pushes to master or release/*
+    branches:    
+      - master
+      - release/*
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+jobs:
+  run-tests:
+    runs-on: self-hosted
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Cluster
+      run: |
+        kind delete cluster 2>/dev/null
+        kind create cluster
+        kubectl get nodes
+    - name: Deploy From Source
+      run: |
+        skaffold run
+    - name: Wait For Pods
+      timeout-minutes: 20
+      run: |
+        set -x
+        kubectl wait --for=condition=available --timeout=500s deployment/adservice
+        kubectl wait --for=condition=available --timeout=500s deployment/cartservice
+        kubectl wait --for=condition=available --timeout=500s deployment/checkoutservice
+        kubectl wait --for=condition=available --timeout=500s deployment/currencyservice
+        kubectl wait --for=condition=available --timeout=500s deployment/emailservice
+        kubectl wait --for=condition=available --timeout=500s deployment/frontend
+        kubectl wait --for=condition=available --timeout=500s deployment/loadgenerator
+        kubectl wait --for=condition=available --timeout=500s deployment/paymentservice
+        kubectl wait --for=condition=available --timeout=500s deployment/productcatalogservice
+        kubectl wait --for=condition=available --timeout=500s deployment/recommendationservice
+        kubectl wait --for=condition=available --timeout=500s deployment/shippingservice
```

Comment:
now here's what I'm thinking, if we add a deployment, we have to add it here, too.

instead can we do:


```
for d in $(kubectl get deployments -o=name); do
  kubectl wait "$d"
done
```

since we expect all deployments to succeed?

Reply:
That is an option as well. I thought it might be better to be explicit which services we're expecting in case one was accidentally removed (until we have more thorough tests in place). I'm fine with either method though